pkg/3scale/amp/operator/redis_reconciler.go

```diff
@@ -56,6 +58,26 @@ func (r *RedisReconciler) Reconcile() (reconcile.Result, error) {
 		return reconcile.Result{}, err
 	}
 
+	// We want to reconcile redis-conf ConfigMap before Deployment
+	// to avoid restart redis pods twice in case of User change ConfigMap.
+	// Annotation GenerationID was added to Pod Template to support Upgrade scenario.
+	// this GenerationID is taken from ConfigMap resourceVersion.
+	// If User change Config Map, Operator will revert it to original one,
+	// but resourceVersion could be changed twice (after user change and after operator).
+	// To avoid this scenario by placing CM reconciliation before Deployment
+	if r.ConfigMap != nil {
+		redisConfigMap := r.ConfigMap(redis)
+		isInternalRedis := r.isInternalRedis()
```

Comment:
no need to check if redis is internal. The redis reconciler is triggered only when redis is internal.

Reply:
Done. Yes, I see now `Internal: operator.NewSystemRedisDependencyReconciler,`. Thank you